package.json

```diff
@@ -58,5 +59,8 @@
     "mathbox": "^2.1.4",
     "react": "^17.0.2 || ^18.0.0",
     "three": ">=0.118.0"
+  },
+  "dependencies": {
+    "lodash": "^4.17.21"
```

Comment:
rather than the full lodash dependency, you might consider https://www.npmjs.com/package/lodash.isequal. I ripped lodash out of threestrap to slim down the build, and the modularized option seemed like a nice way to go too.

Here was another light implementation we could crib with attribution, vs taking a dependency: https://github.com/mapbox/mapbox-gl-js/blob/7e87d7c05c541f98e06df2bc08b96e931ace9549/src/util/util.js#L292-L314

See the discussion here: https://github.com/mapbox/mapbox-gl-js/issues/5744

Maybe it is not good enough but the options maps are simple enough I bet it is.